homeassistant/components/valve/__init__.py

```diff
@@ -186,9 +186,10 @@ def state(self) -> str | None:
 
     @final
     @property
-    def state_attributes(self) -> dict[str, Any]:
+    def state_attributes(self) -> dict[str, Any] | None:
         """Return the state attributes."""
-
+        if not self.reports_position:
+            return None
```

Comment:
Nowadays we always want to include all state attribute items. When a value isn't known it's set to `None`.

Reply:
Yes, but does that apply for an entity which doesn't support some state attribute item?

Reply:
Maybe you're right that it's not expected to be there if it's not supported.

https://github.com/home-assistant/architecture/discussions/680

Although with supported features possibly changing during entity lifetime, the case is a bit muddy.

Reply:
That is a muddy case indeed, but assumingly that is a less commonly happening. But something to discuss for sure 👍